src/stratigraphy/depthcolumn/depthcolumn.py

```diff
@@ -148,8 +203,11 @@ def identify_groups(
 
             matched_blocks = interval.matching_blocks(description_lines, line_index, next_interval)
             line_index += sum([len(block.lines) for block in matched_blocks])
-            groups.append({"depth_intervals": [interval], "blocks": matched_blocks})
-
+            groups.append(
+                # TODO: This seems to be the only case where a list is passed and most of the time it is a list of one
+                # element. Seem to need the function: transform_groups().
```

Comment:
In the identify_groups implementation of the BoundaryDepthColumn subclass, we definitely have some cases were we pass a list of more than one depth interval.

However, I agree with you that we need to clean up the IntervalBlockGroup class, as it is very confusing to allow both a list as well as a single element for the parameters. Either we only allow a list, or we create separate classes for when we need a list and when we need a single element. I'm not sure yet, what would be the best option.